checkov/common/util/var_utils.py

```diff
@@ -0,0 +1,43 @@
+import re
+from typing import Any
+
+from checkov.common.util.parser_utils import find_var_blocks
+
+CFN_VARIABLE_DEPENDANT_REGEX = re.compile(r"(?:Ref)\.[^\s]+")
```

Comment:
Easy win that would catch most CFN cases would be checking for variables with the keyword `Fn::`, and PROBABLY `AWS::` too. I'm not sure regarding `AWS::` because all CFN types have it as a prefix so I don't really remember or not really sure if we'll break anything if we add it..

Reply:
@Saarett We're not implementing the CFn logic yet as it is not too straightforward. Can you open n issue detailing all the cases you can think of, that we should take into account?